x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/permission/FieldPermissionsTests.java

```diff
@@ -144,50 +144,32 @@ public void testParseFieldPermissions() throws Exception {
 
     // test old syntax for field permissions
     public void testBWCFieldPermissions() throws Exception {
-        String q = "{\"indices\": [ {\"names\": \"idx2\", \"privileges\": [\"p3\"], " +
-                "\"fields\": [\"f1\", \"f2\"]" +
-                "}]}";
-        RoleDescriptor rd = RoleDescriptor.parse("test", new BytesArray(q), true,
-                XContentType.JSON);
-        assertArrayEquals(rd.getIndicesPrivileges()[0].getGrantedFields(),
-                new String[]{"f1", "f2"});
-        assertNull(rd.getIndicesPrivileges()[0].getDeniedFields());
-
-        final String failingQuery = q;
+        final String q = "{\"indices\": [ {\"names\": \"idx2\", \"privileges\": [\"p3\"], " +
```

Comment:
I think we can remove this method now. It doesn't check backwards compatibility anymore (as we remove BWC in this PR) and we can get the assurance that we don't parse roles in 2x format in a unit test we can add in RoleDescriptorTests.java

Reply:
yes, makes sense. Replaced these testBWC* tests with a test case in RoleDescriptorTest.testParse